comp/core/flare/helpers/helpers.go

```diff
@@ -0,0 +1,179 @@
+	// automatically be created.
+	//
+	// Example: CopyFile("/etc/datadog/datadog.yaml", "etc") will create a copy of "/etc/datadog/datadog.yaml"
+	// at "etc/datadog.yaml" at the root of the flare.
+	CopyFileTo(srcFile string, destDir string) error
+
+	// CopyFileToRename copies the content of 'srcFile' to 'destDir' in the flare under a new filename.
+	//
+	// The data is automatically scrubbed of any sensitive informations before being copied.
+	//
+	// 'destDir' is a path relative to the flare root (ex: "path/to/a/dir"). Any necessary directory will
+	// automatically be created.
+	//
+	// Example: CopyFile("/etc/datadog/datadog.yaml", "etc", "new_name.yaml") will create a copy of "/etc/datadog/datadog.yaml"
+	// at "etc/new_name.yaml" at the root of the flare.
+	CopyFileToRename(srcFile string, destDir string, filename string) error
+
+	// CopyDirTo copies files from the 'srcDir' to a specific directory in the flare.
+	//
+	// The path for each file in 'srcDir' is passed to the 'shouldInclude' callback. If 'shouldInclude' returns true, the
+	// file is copies to the flare. If not, the file is ignored.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/dir").
+	//
+	// The data of each copied file is automatically scrubbed of any sensitive informations before being copied.
+	//
+	// Example: CopyDir("/var/log/datadog/agent", "logs", <callback>) will copy files from "/var/log/datadog/agent/" to
+	// "logs/agent/" in the flare.
+	CopyDirTo(srcDir string, destDir string, shouldInclude func(string) bool) error
+
+	// CopyDirTo copies files from the 'srcDir' to a specific directory in the flare.
+	//
+	// The path for each file in 'srcDir' is passed to the 'shouldInclude' callback. If 'shouldInclude' returns true, the
+	// file is copies to the flare. If not, the file is ignored.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/dir").
+	//
+	// The data of each copied file is NOT scrubbed of any sensitive informations before being copied. Only files
+	// already scrubbed should be added in this way (ex: agent logs that are scrubbed at creation).
+	//
+	// Example: CopyDir("/var/log/datadog/agent", "logs", <callback>) will copy files from "/var/log/datadog/agent/" to
+	// "logs/agent/" in the flare.
+	CopyDirToWithoutScrubbing(srcDir string, destDir string, shouldInclude func(string) bool) error
+
+	// PrepareFilePath returns the full path of a file in the flare.
+	//
+	// PrepareFilePath will create the necessary directories in the flare temporary dir so that file can be create, but will
+	// not create the file. This method should only be used when the data is generated by another program/library.
+	//
+	// Example: PrepareFilePath("db-monitoring/db-dump.log") will create the 'db-monitoring' directory at the root of the
+	// flare and return the full path to db-dump.log: "/path/to/the/flare/db-monitoring/db-dump.log".
+	PrepareFilePath(path string) (string, error)
```

Comment:
I'd like to avoid any "custom behavior" in the function bases on the parameters.  The key difference here is that `PrepareFilePath` does not create the file ! This is important, has the purpose of this is for another tool/lib to create the required file. We don't control those side tool/lib and they might not accept to overwrite a existing file.

This is required by the flare logic on Windows, for now.